app/views/profiles/edit.html.erb

```diff
@@ -90,9 +90,17 @@
     <h2><%= t '.mfa.multifactor_auth' %></h2>
     <% if @user.mfa_enabled? %>
       <p><%= t '.mfa.enabled' %></p>
-      <%= form_tag multifactor_auth_path, :method => :delete do %>
-        <%= text_field_tag :otp, '', :class => 'form__input' %>
-        <%= submit_tag t('.mfa.disable'), :class => 'form__submit' %>
+      <%= form_tag multifactor_auth_path, :method => :put, :id => 'mfa-edit' do %>
+        <%= label_tag :level, 'Auth Level', :class => 'form__label' %>
+        <%= select_tag :level, options_for_select([
+          ['Login and Write', 'mfa_login_and_write'],
+          ['Login Only', 'mfa_login_only'],
+          ['Disabled', 'no_mfa']], @user.mfa_level), :class => 'form__input form__select' %>
```

Comment:
`Login only` and `Login and write` don't make much sense to me. At least to user, we should show something more appropriate like `UI Authentication` and `UI and API Authentication`.

Reply:
We should also consider updating function names as well. I am not a stickler for naming it `UI and API Authentication` but `..login_and_write` is just misleading. For example, api request to show key is not a write operation and  by `login_only` we don't mean api login. Distinction has more to do with whether we are checking something for api or ui. 
Sorry, I failed to point it out previously.

Reply:
The name _login and write_ or _login only_ derives from the original RFC issue about multi-factor authentication. NPM uses these names to identify their different authentication level. These are basic elements in mfa implementation, so I just borrowed them at that time without too much hesitation.

Currently, fetching API key requires OTP only when auth level set to _login and write_. Truly, these would be confusing to both users and contributors who see this code. Some of other actions not wrapped with mfa now, like API key reset or reset password, but they are also critical. Should they require OTP if _UI Authentication_ enabled? Except that, I think name around _UI_ and _API_ is good. Thanks.

Reply:
I did it. `Auth for UI Only` and `Auth for UI and API` is okay.